### PR TITLE
Fix failing test

### PR DESCRIPTION
**Pull request by johnjosephhorton**
There were a few issues: 

1. We needed to add to Jobs.py code to create the ".edsl_cache.db"
2. The "store" command for cache entry was converting parameters to string (which was stopping job running)
3. Job running fails "show up" in this interview test failure because jobs that are failing for any reason (such as above) cause this test to fail- it's a bit complex and hard to explain. We probably need to re-factor this to show these early exceptions as these are now hard to debug. 

@ellipsis - can you summarize code changes?  

----

| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit b36628eeb94399ea67238e34910966077eb941a4.  | 
|--------|--------|

### Summary:
This PR addresses several issues, makes improvements to the `Cache` and `CacheEntry` classes, updates corresponding test files, and modifies the `CacheEntry` class to correctly handle parameters in the cache entry key generation.

**Key points**:
- Improved the `Cache` and `CacheEntry` classes by adding type hints, improving docstrings, refactoring some methods.
- Fixed issues with the 'Jobs.py' code and the 'store' command for cache entry.
- Addressed job running failures.
- Updated `test_CacheEntry.py` and `test_LanguageModel.py` to reflect changes in the main codebase.
- Modified `gen_key` method in `CacheEntry` class to convert `parameters` to a JSON string before generating the key.
- Updated expected keys in `test_CacheEntry_gen_key` and `test_CacheEntry_key_property` tests to reflect the change in key generation.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: edsl/data/Cache.py [hunk 1]
```diff
@@ -707,67 +635,162 @@ def coop_backup_enabled(value) -> bool:
...
+    def __eq__(self, other_cache: "Cache") -> bool:
+        """
+        Checks if two Cache objects are equal.
+        - Doesn't verify their values are equal, just that they have the same keys.
+        """
+        if not isinstance(other_cache, Cache):
+            return False
+        return self.data.keys() == other_cache.data.keys()
+
+    def __add__(self, other: "Cache"):
```
[the remark this answers is not in the document]
Comment (johnjosephhorton) [reply edsl/data/Cache.py:693]: This is an interesting suggestion, @ellipsis, but this would make a new Cache w/o a sqlite backend. We need to think about this one more. Might just not allow __ add __ here

**Comment by johnjosephhorton** [comment]
Ok @apostolosfilippas this should fix that issue you had. It does surface a bigger issue we have in the unit testing, re: failures caused by "simple"bugs that only show up in the Interview test in this fairly cryptic way

**Review by apostolosfilippas — APPROVED** [review]
